pinot-segment-local/src/main/java/org/apache/pinot/segment/local/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -527,7 +530,7 @@ private void updateDictionary(GenericRow row) {
       IndexContainer indexContainer = entry.getValue();
       Object value = row.getValue(column);
       MutableDictionary dictionary = indexContainer._dictionary;
-      if (dictionary != null) {
+      if (dictionary != null && value != null) {
```

Comment:
avoids NPE on null value in GenericRow

Reply:
I added a test case to provoke this, if null gets into the `GenericRow` and it's not removed, an NPE is thrown here. Since this can and does cause index corruption, I would prefer to be prudent here even if it may be redundant in some cases.

Reply:
Agree we should avoid NPE here. Let's add a separate check for `value != null` and log an error + emit a metric when it happens, then skip the value processing

Reply:
This has been addressed.